flecsi/data/mpi/policy.hh

```diff
@@ -14,563 +14,36 @@
-    return {c, r};
-  }
-  static std::size_t get_row(const Value & v) {
-    return v.first;
-  }
-  static prefixes_base::row get_size(const Value & v) {
-    return v.second;
-  }
-
-  borrow(region_base & r,
-    const partition & p,
-    field_id_t f,
-    completeness = incomplete)
-    : partition(r) {
-    const auto s = p.get_storage<const Value>(f);
-    switch(s.size()) {
-      case 0:
-        break;
-      case 1: {
-        auto & v = s[0];
-        flog_assert(!v.second || v.first == run::context::instance().color(),
-          "sorry: MPI backend does not implement cross-color access");
-        nelems = v.second;
-        break;
-      }
-      default:
-        flog_fatal("underlying partition has size " << s.size() << " > 1");
-    }
+  void init(size_t i) {
+    data[i];
   }
-};
 
-struct intervals {
-  using Value = subrow; // [begin, end)
-  static Value make(subrow r, std::size_t = 0) {
-    return r;
+  local::detail::storage & get_storage(size_t i) {
+    return data.at(i);
   }
 
-  intervals(prefixes & pre,
-    const partition & p,
-    field_id_t fid, // The field id for the metadata in the region in p.
-    completeness = incomplete)
-    : r(&pre.get_base()) {
-    // Called by upper layer, supplied with a region and a partition. There are
-    // two regions involved. The region for `r` stores real field data
-    // (e.g. density, pressure etc.) as the destination of the ghost copy. It
-    // also contains the pairs of (rank, index) of shared entities on remote
-    // peers. The region and associated storage in the partition `p` contains
-    // metadata on which entities are local ghosts (destination of copy) on the
-    // current rank. The metadata is in the form of [beginning index, ending
-    // index), type aliased as Value, into the index space of the entity (e.g.
-    // vertex, edge, cell). We thus need to use the p.get_storage() to get the
-    // metadata, not the region.get_storage() which gives the real data. This
-    // works the same way as how one partition stores the number of elements in
-    // a particular 'shard' on a rank for another partition. In addition, we
-    // also need to copy Values from the partition and save them locally. User
-    // code might change it after this constructor returns. We can not use a
-    // copy assignment directly here since metadata is an util::span while
-    // ghost_ranges is a std::vector<>.
-    ghost_ranges = to_vector(p.get_storage<Value>(fid));
-    // Get The largest value of `end index` in ghost_ranges (i.e. the upper
-    // bound). This tells how much memory needs to be allocated for ghost
-    // entities.
-    if(auto iter = std::max_element(ghost_ranges.begin(),
-         ghost_ranges.end(),
-         [](Value x, Value y) { return x.second < y.second; });
-       iter != ghost_ranges.end()) {
-      max_end = iter->second;
-    }
-  }
+  // this is never called for the MPI backend
+  void get_future(size_t i);
```

Comment:
Removing the declaration causes compilation issues as the functions invoking `get_future` are not templatized, i.e. are always instantiated (even if never called in case of MPI).